### PR TITLE
fix(infra): add timeout to gh CLI token provider

### PR DESCRIPTION
**Pull request by gblanc-1a**
## Description

Fixes a test timeout failure in the CI build where the gh CLI token provider was hanging indefinitely during test execution. Adds a 3-second timeout to prevent the gh CLI token command from hanging, ensuring the authentication flow remains responsive even when gh is slow or unresponsive.

**Related build failure:** https://github.com/AmadeusITGroup/ai-primitives-hub/actions/runs/29904189574/job/88872584613?pr=337

## Type of Change

- [x] 🐛 Bug fix (non-breaking change which fixes an issue)

## Changes Made

- Added `GH_CLI_TIMEOUT_MS` constant (3 seconds)
- Updated `GhCliTokenProvider` constructor to apply timeout to the `exec` call
- This prevents indefinite hangs when the gh CLI command is unresponsive (fixes test timeout: "Timeout of 5000ms exceeded")

## Testing

- [x] Fixes the failing test timeout in CI
- [x] Existing unit tests pass with timeout
- [x] Manual testing completed

## Checklist

- [x] My code follows the project's style guidelines
- [x] I have performed a self-review of my code
- [x] My changes generate no new warnings or errors
- [x] New and existing unit tests pass locally with my changes

## Additional Notes

This fix ensures the CLI and future VS Code extension remain responsive during authentication attempts, even in edge cases where the gh CLI is slow or hanging. The timeout prevents test suites from hanging indefinitely when the gh CLI command is unresponsive.

### CODE REVIEW

**Comment by gblanc-1a** [comment]
### Code review

Found 1 issue:

1. Missing documentation update for authentication timeout behavior (AGENTS.md says "After implementing features or fixing bugs, you MUST update documentation")

https://github.com/anthropics/claude-code/blob/a6ed7ceeb99588262a4045778c318da4c4a698a4/packages/infra/src/auth/gh-cli-token-provider.ts#L1-L20

🤖 Generated with [Claude Code](https://claude.ai/code)

<sub>- If this code review was useful, please react with 👍. Otherwise, react with 👎.</sub>

File: packages/infra/src/auth/gh-cli-token-provider.ts [hunk 1]
```diff
@@ -32,8 +32,12 @@ import {
 
 export type ExecFn = (command: string) => Promise<{ stdout: string }>;
 
+const GH_CLI_TIMEOUT_MS = 3000;
```
Comment (mvgadagi) [inline packages/infra/src/auth/gh-cli-token-provider.ts:35]: Hey @gblanc-1a, thanks for the proposal and bringing the doc update ❤️ , It's really good to ensure timeout. 
But I am worried about the 3s being too aggressive, assuming we invoke `gh auth token` process command sometime this itself have some latency and promise will resolve much before process kicks in.
Do you think we go for 10s to start? But I approved the PR if you think its fine to leave 3s

**Review by mvgadagi — APPROVED** [review]

**Review by wherka-ama — APPROVED** [review]
/LGTM: it makes total sense indeed! Thank you ❤️

File: packages/infra/src/auth/gh-cli-token-provider.ts (hunk 1 above)
Comment (wherka-ama) [reply packages/infra/src/auth/gh-cli-token-provider.ts:35]: If it's only about getting the token, the operation should be almost instant. Then we may hit the token rotation, which may indeed take longer. 
However, this timeout value is quite arbitrary. It doesn't really matter if it's 3s or 10. If the network is bad we may still see it. Sometimes it's almost better to fail quickly and flag it to the user that there are some issues with the network.

Unless we notice actual reports from the users claiming it happens a lot.

On that note: I'm not sure if we are properly handling the cases where people hit the API rate limits and if we propagate such errors to the users. It would be good to tackle it as well as that may actually happen even more often.

**Review by wherka-ama — APPROVED** [review]

File: packages/infra/src/auth/gh-cli-token-provider.ts (hunk 1 above)
Comment (mvgadagi) [reply packages/infra/src/auth/gh-cli-token-provider.ts:35]: Okay Waldek sure, probably I am wrong, I was thinking like since exec method to invoke command itself take some some time, we will be left with less time to get the token, now since we all aligned we will monitor and probably extend this to rate limits etc like you added.
